packages/cli/src/lib/commands/collect/utils/persist/persist-flow.unit.test.ts

```diff
@@ -0,0 +1,123 @@
+    jest.spyOn(flow, 'createFlowResult').mockResolvedValue({mock: 'jsonResult'});
+    await persistFlow(flow, { outPath: '', format: ['json'], url: 'mock.com' });
+    expect(writeFile).toHaveBeenCalledWith('report.json', JSON.stringify({mock: 'jsonResult'}));
+  });
+
+  it('should return the path to the json report if json is given as format', async () => {
+    jest.spyOn(fs, 'existsSync').mockReturnValue(true);
+    const reports = await persistFlow(flow, { outPath: '', format: ['json'], url: 'mock.com' });
+    expect(reports).toStrictEqual([expect.stringContaining('.json')]);
+  });
+
+  it('should extract an html report from UserFlow if html is given as format', async () => {
+    const generateReportSpy = jest.spyOn(flow, 'generateReport');
+    await persistFlow(flow, { outPath: '', format: ['html'], url: 'mock.com' });
+    expect(generateReportSpy).toHaveBeenCalled();
+  });
+
+  it('should save the report in html if html is given as format', async () => {
+    jest.spyOn(fs, 'existsSync').mockReturnValue(true);
+    jest.spyOn(flow, 'generateReport').mockResolvedValue('Mock HTML Report');
+    await persistFlow(flow, { outPath: 'Flow', format: ['html'], url: 'mock.com' });
+    expect(writeFile).toHaveBeenCalledWith(join('Flow', 'report.html'), 'Mock HTML Report');
+  });
+
+  it('should return the path to the html report if html is given as format', async () => {
+    jest.spyOn(fs, 'existsSync').mockReturnValue(true);
+    const reports = await persistFlow(flow, { outPath: '', format: ['html'], url: 'mock.com' });
+    expect(reports.at(0)).toContain('.html');
+  });
+
+  it('should extract an md report from the json report if md is given as format', async () => {
+    jest.spyOn(flow, 'createFlowResult').mockResolvedValue({mock: 'base for md report'});
+    const createReducedReportSpy = jest.spyOn(reportUtils, 'createReducedReport').mockReturnValue({mock: 'reduced report'} as any as ReducedReport);
+    const generateMdReportSpy = jest.spyOn(mdReportUtils, 'generateMdReport');
+    await persistFlow(flow, { outPath: '', format: ['md'], url: 'mock.com' });
+    expect(createReducedReportSpy).toHaveBeenCalledWith({mock: 'base for md report'})
+    expect(generateMdReportSpy).toHaveBeenCalledWith({mock: 'reduced report'});
+  });
+
+  it('should save the report in md if md is given as format', async () => {
+    jest.spyOn(fs, 'existsSync').mockReturnValue(true);
+    jest.spyOn(mdReportUtils, 'generateMdReport').mockReturnValue('Mock Md Report')
+    await persistFlow(flow, { outPath: 'Flow', format: ['md'], url: 'mock.com' });
+    expect(writeFile).toHaveBeenCalledWith(join('Flow', 'report.md'), 'Mock Md Report');
+  });
+
+  it('should return the path to the markdown report if md is given as format', async () => {
+    jest.spyOn(fs, 'existsSync').mockReturnValue(true);
+    const reports = await persistFlow(flow, { outPath: '', format: ['md'], url: 'mock.com' });
+    expect(reports.at(0)).toContain('.md');
```

Comment:
Can there be more than one element in the resulting array?
If not, narrowing the result down to a dynamic value in an array of one element is generally more precise (as it is done for `json`).